torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -413,11 +413,11 @@ void FunctionParameter::set_default_str(const std::string& str) {
     }
   } else if (type_ == ParameterType::LAYOUT) {
     if (str == "None") {
-      default_layout = nullptr;
+      // do nothing
```

Comment:
This is a memory hazard, as when `ParameterType::LAYOUT`, `default_layout` may not be initialized. Maybe add an assert here that it is optional, in this case.

Reply:
(There's actually a lot of hazards here, so https://github.com/pytorch/pytorch/pull/33771 was sort of trying to tighten things up, but I also ran out of time on that PR too :)

Reply:
Fixed with assert allows_none